install.sh

```diff
@@ -17,58 +14,161 @@ CYAN='\033[0;36m'
 ### clear colors
 NC='\033[0m'
 
-if [ -f "fm6000.pl" ]; then
-    printf '%b\n' "${BLUE}Seems like you cloned the repository${NC}"
-    cp fm6000.pl fm6000
+out() {
+	# Print a message
+	printf '%b\n' "$@"
+}
+
+notset() {
+	case $1 in '') return 0 ;; *) return 1 ;; esac
+}
+
+check_dep() {
+	if [ "$(command -v "$1")" ]; then return 0; else return 1; fi
+}
+
+## get options
+for opt in "$@"; do
+	case $opt in
+	--nocolors | -nc)
+		# unset colors
+		for i in RED GREEN YELLOW BLUE CYAN NC; do
+			unset $i
+		done
+		;;
+	--root | -r)
+		root=1
+		;;
+	--no*root | -nr)
+		root=0
+		;;
+	--install*path=*)
+		install_path="${opt#*=}"
+		;;
+	--dry*run | -dr)
+		dryrun=1
+		;;
+	--headless | -y)
+		headless=1
+		;;
+	-h | *)
+		[ -x "./install.sh" ] && prog="./install.sh" || prog="sh -c \"\$(curl https://raw.githubusercontent.com/anhsirk0/fetch-master-6000/master/install.sh)\""
+		out \
+			"${RED}------${NC} ${CYAN}Fetch-Master-6000 install script${NC} ${RED}------${NC}
+${YELLOW}Usage:${NC} $prog ${BLUE}<options>${NC}
+
+ ${RED}--nocolors${NC},${RED} -nc    ${GREEN}Do not print colored text${NC}
+ ${RED}--root${NC},${RED}     -r     ${GREEN}Use root privileges(it is automatically detected but still can be used)${NC}
+ ${RED}--noroot${NC},${RED}   -nr    ${GREEN}Do not use root privileges(it is automatically detected but still can be used)${NC}
+ ${RED}--dry-run${NC},${RED}  -dr    ${GREEN}Dry run fm6000${NC}
+ ${RED}--headless${NC},${RED} -y     ${GREEN}install headless(without prompting)${NC}
+"
+		return 1
+		;;
+	esac
+done
+
+# make script compatible with env variables
+
+## set install_path
+notset "$install_path" && {
+	if [ -x "$HOME/.local/bin" ]; then
+		install_path=$HOME/.local/bin
+	elif [ -x "/usr/local/bin" ]; then
+		install_path=/usr/local/bin
+	fi
+}
+
+## check if $install_path requires root group
+notset "$install_path" || {
+	[ ! -x "$install_path" ] && {
+		out "${RED}$install_path does not exist${NC}"
+		return 1
+	}
+	check_dep "stat" && {
+		if [ "$(stat -c "%G" $install_path)" = "root" ]; then
+			notset "$root" && root=1
+		fi
+	}
+}
```

Comment:
Check the group of install path

if group equals to root, root privileges will be used